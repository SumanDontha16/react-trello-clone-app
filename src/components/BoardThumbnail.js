import React from "react";
import styled from "styled-components";

const Thumbnail = styled.div`
  height: 40px;
  width: 200px;
  background: black;
  padding: 10px;
  margin: 8px;
  display: flex;
  justify-content: center;
  align-items: center;
  cursor: pointer;
  border-radius: 3px;
`;

const Title = styled.h4`
  color: white;
  text-decoration: none;
`;

const BoardThumbnail = ({ title }) => {
  return (
    <Thumbnail>
      <Title>{title}</Title>
    </Thumbnail>
  );
};

export default BoardThumbnail;
