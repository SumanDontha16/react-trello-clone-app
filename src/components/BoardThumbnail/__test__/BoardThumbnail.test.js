import React from 'react'
import { render, screen } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import BoardThumbnail from '../BoardThumbnail'
import renderer from 'react-test-renderer'

describe('Render Board Thumbnail Component', () => {

  it('Board Thumbnail snapshot', () => {
    const createBoardThumbnail = renderer
      .create(<BoardThumbnail />)
    expect(createBoardThumbnail).toMatchSnapshot()
  })

  it('Should render with correct title', () => {
    render(<BoardThumbnail title="Create a new Board" />)

    const titleE1 = screen.getByTestId("title")
    expect(titleE1.textContent).toBe("Create a new Board")
  })

})
