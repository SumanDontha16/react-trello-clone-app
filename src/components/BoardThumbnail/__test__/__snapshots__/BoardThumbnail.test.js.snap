// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Board Thumbnail Component Board Thumbnail snapshot 1`] = `
<div
  className="sc-bdVaJa OyGdR"
>
  <h4
    className="sc-bwzfXH KUmQF"
    data-testid="title"
  />
</div>
`;
